### PR TITLE
Review request v3 for Microsoft.Portal to onboard to TypeSpec

### PR DESCRIPTION
**Pull request by sayeedalam-microsoft**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [ ] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [x] Other, please clarify:
    - Onboarding Microsoft.Portal to TypeSpec

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories")
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

**Comment by sayeedalam-microsoft** [comment]
@ARM Reviewer 
Please approve and advise with the suppression process. 

Getting Swagger LintDiff error. [TenantLevelAPIsNotAllowed](https://github.com/Azure/azure-openapi-validator/blob/f3fcf47c9f74b15e90e5198bf09bb6569e13213e/docs/tenant-level-apis-not-allowed.md)

We got the exception from the PAS team:
image

**Comment by raosuhas** [comment]
@sayeedalam-microsoft  the exception looks good to me. But if you are upgrading to typespec , then please do it in place for an existing versions. Do not combine a typespec conversion PR with a version upgrade PR

**Comment by sayeedalam-microsoft** [comment]
Breaking change.
https://aka.ms/brch
Intake form: https://msazure.visualstudio.com/One/_workitems/edit/28847237

**Comment by ms-bogdan-rotaru** [comment]
Can you have a look at `Swagger LintDiff` check?

File: specification/portal/Dashboard.Management/Dashboard.tsp [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+/**
+ * The shared dashboard resource definition.
+ */
+@doc("The shared dashboard resource definition.")
+model Dashboard
+  is Azure.ResourceManager.TrackedResource<DashboardPropertiesWithProvisioningState> {
+  ...ResourceNameParameter<
+    Resource = Dashboard,
+    KeyName = "dashboardName",
+    SegmentName = "dashboards"
```
Comment (allenjzhang) [inline specification/portal/Dashboard.Management/Dashboard.tsp:24]: optional

File: specification/portal/Dashboard.Management/Dashboard.tsp [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+namespace Microsoft.Portal;
+/**
+ * The shared dashboard resource definition.
+ */
+@doc("The shared dashboard resource definition.")
+model Dashboard
+  is Azure.ResourceManager.TrackedResource<DashboardPropertiesWithProvisioningState> {
+  ...ResourceNameParameter<
+    Resource = Dashboard,
+    KeyName = "dashboardName",
```
Comment (allenjzhang) [inline specification/portal/Dashboard.Management/Dashboard.tsp:23]: optional. auto constructed from `Dashboard`

File: specification/portal/Dashboard.Management/Dashboard.tsp [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  /**
+   * Updates an existing Dashboard.
+   */
+  @parameterVisibility
+  update is ArmCustomPatchSync<Dashboard, PatchableDashboard>;
+
+  /**
+   * Deletes the Dashboard.
+   */
+  delete is ArmResourceDeleteSync<Dashboard>;
```
Comment (allenjzhang) [inline specification/portal/Dashboard.Management/Dashboard.tsp:49]: Should use `ArmResourceDeleteWithoutOkSync`

File: specification/portal/Dashboard.Management/Dashboard.tsp (hunk 1 above)
Comment (sayeedalam-microsoft) [reply specification/portal/Dashboard.Management/Dashboard.tsp]: Removed

File: specification/portal/Dashboard.Management/Dashboard.tsp (hunk 2 above)
Comment (sayeedalam-microsoft) [reply specification/portal/Dashboard.Management/Dashboard.tsp]: Removed

File: specification/portal/Dashboard.Management/Dashboard.tsp (hunk 3 above)
Comment (sayeedalam-microsoft) [reply specification/portal/Dashboard.Management/Dashboard.tsp:47]: Do you mean `ArmResourceDeleteWithoutOkAsync`? But the existing one is not `async`. Thus, keeping it as it is.

**Review by msHich — APPROVED** [review]

**Comment by allenjzhang** [comment]
## Sorry for the delay. But here is my analysis on the flagged cross version breaking change:

### 1009 - RemovedRequiredParameter
    
    **Actionable:**
    
    The required parameter 'dashboard' was removed in the new version.
    https://github.com/Azure/azure-rest-api-specs/blob/main/specification/portal/resource-manager/Microsoft.Portal/preview/2020-09-01-preview/portal.json#L101:11
    
    If you are still using autorest:
    You can use following to keep swagger same
    
    `@@encodedName(Dashboards.createOrUpdate::parameters.resource, "application/json", "dashboard");`
    
    AND
    
    You should use @clientName to keep the body param same in old SDK. https://azure.github.io/typespec-azure/docs/howtos/migrate-swagger/migrate-arm-tips#changing-the-names-of-request-payload-parameters
    
### 1010 - AddingRequiredParameter
    
    The required parameter 'properties' was added in the new version.
    [Microsoft.Portal/preview/2022-12-01-preview/portal.json#L515:11](https://github.com/Azure/azure-rest-api-specs/blob/c0a799df9939935eca1b6db1eb6bd9ed55c7468c/specification/portal/resource-manager/Microsoft.Portal/preview/2022-12-01-preview/portal.json#L515:11)
    
### 1017 - ReferenceRedirection
    
    - There are a few such as error can be ignored, since old custom ErrorDefinition is standardized to ErrorDetail.
    - There are some due to rename such as `DashboardPropertiesWithProvisioningState` vs `DashboardProperties`. You would need to decide whether to keep the same or incurr SDK changes. Again, @clientName can be used to shape the SDK while @encodedName to shape swagger
    - 
    
### 1022 - RemovedAdditionalProperties
    
    **Actions Needed**
    
    You should use `metadata? : Record<unknown>` instead of `metadata? : {}`.
    
### 1023 - TypeFormatChanged
    
    Ignore This may create SDK changes. But it may be right to go to common types v5 standard over custom forked common definitions.
    
### 1025 - RequiredStatusChange
    
    Ignore
    
### 1026 - TypeChanged
    
    **Action/Review needed**
    -  The new version has a different type 'boolean' than the previous one 'string'.
    -  The new version has a different type 'string' than the previous one 'integer'.
    
### 1029 - ReadonlyPropertyChanged
    
    Ignore: Possbile SDK impact, but can't fix
    
### 1032 - DifferentAllOf
    
    Ignore: Standardizing to common-type via allof
    
### 1034 - AddedRequiredProperty
    
    Ignore: page.value is now required but it is a response model.
    
### 1036 - ConstraintChanged
    
    Ignore: Refinement on the dashboardName regEx patterns. Non-sdk impact anyway.
    
### 1038 - AddedPath
    
    Ignore: I assume new version added new `tenantConfiguration` operations was intentional
    
### 1040 - AddedReadOnlyPropertyInResponse
    
    Ignore: You have redefined standard types in your swagger instead of ref to common-types. They are removed now.
    
### 1048 - AddedXmsEnum
    
    Ignore: Refinement of string type into extensible string based enum
    
### 1007 - RemovedClientParameter
    
    Ignore:  TypeSpec inline the parameter
    
### 1012 - RemovedResponseCode
    
    Ignore. No need to declare standard 404 
    
### 1045 - AddedOptionalProperty
    
    Ignore: Old definition was not up to date.

**Review by rkmanda — APPROVED** [review]

**Comment by danielrbradley** [comment]
@sayeedalam-microsoft please could you clarify if the lense part metadata type can really only be `Extension/HubsExtension/PartType/MarkdownPart` or if there are other possible valid values such as `Extension/HubsExtension/PartType/ArgQuerySingleValueTile`, `Extension/HubsExtension/PartType/ArgQueryChartTile` and `Extension/HubsExtension/PartType/ArgQueryGridTile`?

Could we add all possible valid values to the specification, or otherwise broaden this field to be of type `string` to avoid flagging valid fields as errors?

**Pull request merged**